### ISSUE TITLE
F-Droid

### ISSUE THREAD

**Issue by Poussinou** [ulan17/TimeTable#13]
Hi,

Since this app is GNU GPL v3 licensed, are you considering the fact to add it to [F-Droid](https://f-droid.org/)? There is a strong community behind.

You can find some documentation [here](https://f-droid.org/docs/).

Otherwise, if you don't want to work on the inclusion of this app by yourself, would it be ok for you if someone else do it for you?

**Comment by ulan17** [comment]
Hi, @Poussinou. I think, app isn't completed to add F-Droid. There is too much stuff that I have to work on them.

**Comment by Poussinou** [comment]
No problem, just ping me when you think it's ok ;)

**Comment by IzzySoft** [comment]
If you think it's not "fully ready" for the official repo, but "ready enough" that some folks should be able to test it and report back to you, helping you to improve – there's also [my repo](https://apt.izzysoft.de/fdroid/repo) which already served as stepping stone for many apps (73 at this time of writing, which meanwhile rreached the official repo – and several still use it for their beta versions in exactly this manner). If that appeals you, simply attach an APK to the corresponding tag (ugh, of course that means you should use tags – which in the end you'd need for the official F-Droid repo anyway) and ping me, I'd then pick it up.

**Comment by flxholle** [comment]
> Hi, @Poussinou. I think, app isn't completed to add F-Droid. There is too much stuff that I have to work on them.

Hi,
is this repository still under active development?
I created a Fork (https://github.com/asdoi/TimeTable), in which I included some more features, and I would like to publish the fork to f-droid.
Also I created a pull request and if you decide to merge my changes, there's no need for me to publish my fork.

**Comment by IzzySoft** [comment]
@asdoi just in case: if you use a different package name, and then start tagging releases plus providing the (signed) APK, I could immediately add your fork to my repo while we wait for a final decision here. Just ping me when I should pick it up.

**Comment by flxholle** [comment]
@IzzySoft 
Thank you very much.
I created a new branch here:
https://github.com/asdoi/TimeTable/tree/izzyOnDroidRepo
and added a Release. I also added some more screenshots.
Should I add a fastlane structure for description and screenshots?

**Comment by IzzySoft** [comment]
Thanks @asdoi – and yes, Fastlane would be much appreciated! Should @ulan17 decide to pick up again, that could go here via a separate PR as well. F-Droid will want it anyway, so it's not "wasted time".

**Comment by IzzySoft** [comment]
PS: Will [show up here](https://apt.izzysoft.de/fdroid/index/apk/com.asdoi.timetable) in about 3 hours. I'll update description etc. once Fastlane is available.

**Comment by flxholle** [comment]
Thank you very much @IzzySoft 
I created the fastlane structure.

**Comment by IzzySoft** [comment]
Thanks @asdoi – comparing now to see what I need to update on my end (app is already live btw). Pointers:

* `short_description.txt`: limited to 80 chars, all behind that will be truncated. Yours has 93 chars, hence the last 13 will "disappear" (summary will end with "homewor"). Suggestion: As it's clear to be an Android Application (even from the Fastlane path), that can be skipped here. I truncated it to "app for students to save timetable, homeworks and notes" (though "app" is obsolete as well, it sounds weird without; maybe "Students can save timetable, homeworks and notes", or "for students to save timetable, homeworks and notes"?)
* `full_description.txt`: you can use HTML tags here if you want – e.g. for the link and for the list. But it looks fine as-is, so this is purely optional.
* screenshots: nice, no frames – I love that :smiley: Though I'd "half the size" (think of mobile data when viewing the catalog); 720x1480 should be big enough (for my repo, I even go down for 350x*).

Missing a featureGraphic and the per-release changelogs – but they are of course purely optional. With them, your app would be present on the initial page in the F-Droid app :wink: Should you add that please let me know so I fetch it. Changelogs would be fetched automatically with the next release then by my updater as well.

**Comment by flxholle** [comment]
Thanks @IzzySoft 
I updated the description and added featureGraphic.
Furthermore the inclusion criteria on F-Droid include the agreement of the developer.
I don't know if a fork matches this (or isn't this handled so strictly)

**Comment by IzzySoft** [comment]
Thanks! I've added the featureGraphic (nice one!), will show up tomorrow.

And your fork will be perfectly acceptable at F-Droid: it matches with the license, and you changed the package name – so no "transfer of ownership" is needed (disclosure: I'm one of the maintainers there). Go ahead and apply if you wish, or give @ulan17 a few days to consider. Choice is yours :wink:

**Comment by flxholle** [comment]
Hi @IzzySoft 
Yesterday my app has been published on F-Droid: https://f-droid.org/en/packages/com.asdoi.timetable/
I think now you can exclude the app for future updates in your repo.
Thank you very much for your help.

**Comment by IzzySoft** [comment]
Thanks for the reminder! I usually give apps a few days in parallel (so users get their indexes updated). Have a script here in place cross-checking and telling me which apps can be removed from my repo (you couldn't know that). So good I've heard it from you directly, and then can remove it with good conscience (sometimes people want their app to stay for different reasons, which is why you'll always see a little overlap).

But I think this issue can be closed now, right?
